### ISSUE TITLE
Shouldn't force quoted identifiers

### ISSUE THREAD

**Issue by kylebutler-sf** [dbt-labs/dbt-utils#644]
### Describe the bug

`Adapter.quote` is used in many places in this package which forces quoted identifiers. My specific use case is with "star" in which case it will quote all of the columns. Depending on the target system this can cause quoted identifiers to be required to query columns generated by dbt. 

### Steps to reproduce

```
with source as (

    select * from {{ source('_lenel__sources', 'accesspane') }}

),

renamed as (

    select
        name as panel_name,
        panelid as panel_id,
        daylightsave as daylight_savings,
        paneltype as panel_type,
        panelnum as panel_number,
        paneltz as panel_timezone,
        audit_etl_job_ins_key as etl_insert_key,
        audit_etl_job_upd_key as etl_update_key,
        audit_etl_job_ins_ts as etl_inserted_at,
        audit_etl_job_upd_ts as etl_updated_at,
        table_processing_datetime as table_processed_at,
        {{ dbt_utils.star(
            from=source('_lenel__sources', 'accesspane'),
            except=["name", "panelid", "daylightsave", "paneltype", "panelnum", "paneltz", "audit_etl_job_ins_key", "audit_etl_job_upd_key", "audit_etl_job_ins_ts","audit_etl_job_upd_ts","table_processing_datetime"],
            prefix='panel_'
        )|lower }}

    from source

)

select * from renamed
```

All columns expanded by star will be quoted

### Expected results

We would expect this to respect the settings of the project.
```
baudrate as panel_baudrate,
comport as panel_comport,
terminalna as panel_terminalna,
cardholdsz as panel_cardholdsz,
cardtype as panel_cardtype,
pintype as panel_pintype,
```

### Actual results

``` 
"baudrate" as "panel_baudrate",
"comport" as "panel_comport",
"terminalna" as "panel_terminalna",
"cardholdsz" as "panel_cardholdsz",
"cardtype" as "panel_cardtype",
"pintype" as "panel_pintype",
```

### System information
**The contents of your `packages.yml` file:**
```
packages:
  - package: dbt-labs/codegen
    version: 0.7.0
  - package: dbt-labs/dbt_utils
    version: 0.8.6
```

**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [ ] bigquery
- [X] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
➜ dbt --version
Core:
  - installed: 1.1.0
  - latest:    1.2.0 - Update available!

  Your version of dbt-core is out of date!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation

Plugins:
  - bigquery:  1.1.0 - Update available!
  - snowflake: 1.1.0 - Update available!
```

### Are you interested in contributing the fix?

Slightly outside my wheelhouse, though I'm interested in the resolution.

**Comment by joellabes** [comment]
I empathise with the problem you describe! If you or another community member wants to change it to a configurable behaviour, I'd welcome a PR in that direction 👍

**Comment by CR-Lough** [comment]
I got this! Working on it in the Hackathon #Coalesce

**Comment by joellabes** [comment]
Resolved by #706
